components/inputs/input-time.js

```diff
@@ -194,6 +204,9 @@ class InputTime extends LitElement {
 		if (this.label === null) {
 			console.warn('d2l-input-time component requires label text');
 		}
+		if (this.value === undefined) {
+			this.value = formatValue(getDefaultTime(this.defaultValue));
```

Comment:
And the reason we only have to do this the first time is that now this becomes the "last known good" value so it'll revert to it in the future when an invalid value is entered.

Reply:
~Correct. We used to assign `_value` and `_formattedValue` directly in the constructor because they relied on an constant external to the component instance, but as I understand I can't get our dynamic default until `firstUpdated` (after the regular `value` property is already assigned) we only need to overwrite it if no value was assigned.~

Found a flaw with this. If the `value` is set to something invalid in the html the value will be set to midnight instead of undefined before we get here - so if the value is junk then we end up with 12AM instead of the default 🤔 but there's presently no way to tell if what was originally passed in was 12AM or something else. I'd need to see what `value` was raw before it passed through the setter, and I'm not sure how to do that.

Reply:
So seems like you need to apply the default  in the `value` setter? Will `oldValue` initially be `undefined` the first time it runs? Maybe you can use that somehow...

Reply:
Because of the need to check against the default value and given value together and the way they don't both seem to exist when it's called the first time I took a different approach to figure out the initial value after the setter is called. Feedback welcome.